### PR TITLE
Added makefile and updated whether a column could be null

### PR DESCRIPTION
**Pull request by jonathankim345**
## Notion ticket link

[Add Makefile](https://www.notion.so/uwblueprintexecs/Add-Makefile-fe6e8312ecbd4d2d8bfb6c998f83a8db?pvs=4)

## Implementation description
* I created a make file with important commands that included all the required targets for linting, formatting, running migrations, generating migrations, starting/stopping Docker, cleaning the database, and seeding the database, while adding additional functionality for reading all the tables as well as being able to check the data from a specific table.
* I tested every single method to ensure its functionality, but in the process I also reverted my table into an earlier more primitive format, and I had to change the column of email to allow for null values. 
* The final thing involves the seeding target, as the seeding script has not been created yet, it has been left as a comment and it echoes a dummy message when called 

## Steps to test
1.  Verify all targets are functional (all of them should work with make installed)

## Checklist
- [✅] My PR name is descriptive and in imperative tense
- [✅] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [✅ ] I have run the appropriate linter(s)
- [✅] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+format:
+	docker exec -it humane_society_backend /bin/bash -c "yarn fix"
+	docker exec -it humane_society_frontend /bin/bash -c "yarn fix"
+
+# Database Migrations
+migration:
+	docker exec -it humane_society_backend /bin/bash -c "node migrate up"
+
+generate-migration:
+	docker exec -it humane_society_backend /bin/bash -c "node migrate create --name $(name).ts"
```
Comment (jerry-cheng5) [inline Makefile:18]: I think for any of these commands that take in arguments, we should output some error message if no argument is passed.

Also please update the help command with arguments (maybe an example call for the command)

File: backend/typescript/migrations/2024.06.15T17.13.39.add-user-columns.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { Migration } from "../umzug";
 export const up: Migration = async ({ context: sequelize }) => {
   await sequelize.getQueryInterface().addColumn("users", "email", {
     type: DataType.STRING,
-    allowNull: false,
+    allowNull: true,
```
[the remark this answers is not in the document]
Comment (aanxniee) [reply backend/typescript/migrations/2024.06.15T17.13.39.add-user-columns.ts:7]: i dont think it should be nullable

**Review by aanxniee — COMMENTED** [review]
agree with jerrys comments, you can unstage your migration changes. only the makefile should be in this pr

File: backend/typescript/migrations/2024.06.15T17.13.39.add-user-columns.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (aanxniee) [reply backend/typescript/migrations/2024.06.15T17.13.39.add-user-columns.ts:7]: okay i see why u made it nullable, rn it'll give you an error bc our entries don't have emails in them. i'll go over this in today's sync

[the remark this answers is not in the document]
Comment (jonathankim345) [reply backend/typescript/migrations/2024.06.15T17.13.39.add-user-columns.ts]: Yes for sure! I changed this thank you for the feedback

File: Makefile (hunk 1 above)
Comment (jonathankim345) [reply Makefile:21]: Great suggestions! I made sure to add them :D it was a good call to update the help command as well

**Pull request merged**